src/js/providers/tracks-mixin.ts

```diff
@@ -985,4 +990,17 @@ function cuesMatch(cue1: TrackCue, cue2: TrackCue): boolean {
         JSON.stringify(cue1.value) === JSON.stringify(cue2.value);
 }
 
+// Sets old caption tracks to parsed = true to avoid accumulating captions across playlist items.
+function hideIfOldCaptionsSafari(tracks: TracksRecord, track: TextTrackLike, trackId: string): boolean {
+    if (Browser.safari  && track.kind ==='captions') {
+        if (track.parsed && !tracks[trackId]) {
+            track.mode = 'hidden';
```

Comment:
Shouldn't mode be set to `disabled` if the track is not in use? We will still get events for hidden tracks.

Reply:
That's correct, just pushed a new commit with the change.